Makefile

```diff
@@ -24,37 +26,58 @@ sedseparator = ''
 sedignorecase =
 endif
 
-build_dir = 'tmp/_output'
+build_dir := build/_output
 configcli_dest := $(build_dir)/configcli.tgz
 
 .DEFAULT_GOAL := build
 
 build: configcli pkg/apis/kubedirector.bluedata.io/v1alpha1/zz_generated.deepcopy.go | $(build_dir)
-	@echo
 	@echo \* Creating KubeDirector deployment image and YAML...
 	@test -d vendor || dep ensure -v
-	@echo operator-sdk build ${image}
-	@operator-sdk build ${image} | grep -v "Create deploy/operator.yaml"
+	operator-sdk build ${image}
 	@docker image prune -f > /dev/null
-	@sed -i ${sedseparator} \
-        -e '/command:/ {' \
-        -e 'n; ' \
-        -e 's~.*~          - "/bin/sh"~; G; ' \
-        -e 's~$$~          args:~; G; ' \
-        -e 's~$$~          - "-c"~; G; ' \
-        -e 's~$$~          - "mkfifo /tmp/fifo; (/root/${project_name} \&> /tmp/fifo) \& while true; do cat /tmp/fifo; done"~;' \
-        -e '}' deploy/operator.yaml
-	@sed -i ${sedseparator} \
+	@sed ${sedseparator} \
+        -e '/command:/ { n; ' \
+        -e 's~.*~          - "/bin/sh"~; ' \
+        -e 'a\          args:' \
+        -e 'a\          - "-c"' \
+        -e 'a\          - "mkfifo /tmp/fifo; (/root/${project_name} \&> /tmp/fifo) \& while true; do cat /tmp/fifo; done"' \
+        -e '}' \
         -e '/env:/ {' \
-        -e 'G; ' \
-        -e 's~$$~            - name: MY_NAMESPACE~; G; ' \
-        -e 's~$$~              valueFrom:~; G; ' \
-        -e 's~$$~                fieldRef:~; G; ' \
-        -e 's~$$~                  fieldPath: metadata.namespace~;' \
-        -e '}' deploy/operator.yaml
+        -e 'a\            - name: MY_NAMESPACE' \
+        -e 'a\              valueFrom:' \
+        -e 'a\                fieldRef:' \
+        -e 'a\                  fieldPath: metadata.namespace' \
+        -e '}' \
+        -e 's~REPLACE_IMAGE~${image}~' \
+        deploy/operator.yaml >${local_deploy_yaml}
+	@echo done
+	@echo
 
-	@echo "      serviceAccountName: kubedirector" >> deploy/operator.yaml
-	@mv deploy/operator.yaml deploy/kubedirector/deployment-localbuilt.yaml
+debug: export GOFLAGS += -gcflags=all=-N -gcflags=-l
+debug: build
+debug:
+	@echo \* Applying debug settings to KubeDirector deployment YAML...
+	@sed -i ${sedseparator} \
+        -e '/\/root\/${project_name}/s~~/root/dlv --listen=:40000 --headless --api-version=2 exec &~' \
+        -e '/template:/,/metadata:/ {' \
+        -e '/metadata:/ {' \
+        -e 'a\      annotations:' \
+        -e 'a\        container.apparmor.security.beta.kubernetes.io/${project_name}: unconfined' \
+        -e '}' \
+        -e '}' \
+        -e '/image:/ {' \
+        -e 'a\          ports:' \
+        -e 'a\          - containerPort: 40000' \
+        -e 'a\            hostPort: 40000' \
```

Comment:
nit: What would it take to make this NodePort instead of hostPort

Reply:
I used NodePort to make debugging convenient. This is the port used to connect to the debugger process within the container. In the debugger UI you need to specify an IP address and a port to connect to. If we used the NodePort we would have to use the container IP address and that's always changing. By using the HostPort we can always use localhost as the IP address, assuming the UI is running on the same node as the container.